File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/communication/message/response/ErrorResponse.java
package lunalauf.rms.utilities.network.communication.message.response;

import com.google.gson.annotations.SerializedName;
import lunalauf.rms.utilities.network.communication.ErrorType;
import lunalauf.rms.utilities.network.communication.message.type.ResponseType;

public class ErrorResponse extends Response {

    @SerializedName("error")
    public ErrorType error;

    public ErrorResponse() {
        super(ResponseType.ERROR);
    }
}

File: utilities/src/main/kotlin/lunalauf/rms/utilities/network/communication/message/response/ErrorResponse.kt
package lunalauf.rms.utilities.network.communication.message.response

import com.google.gson.annotations.SerializedName
import lunalauf.rms.utilities.network.communication.ErrorType
import lunalauf.rms.utilities.network.communication.message.type.ResponseType

class ErrorResponse : Response(ResponseType.ERROR) {
    @SerializedName("error")
    var error: ErrorType = ErrorType.UNKNOWN_ERROR

    override fun toString(): String {
        return super.toString()
            .replace(")", "[$error])")
    }
}
